lib/bolt/node/winrm.rb

```diff
@@ -271,10 +276,37 @@ def _run_command(command)
 
     def _run_script(script, arguments)
       @logger.info { "Running script '#{script}'" }
-      with_remote_file(script) do |remote_path|
-        path, args = *process_from_extension(remote_path)
-        args += escape_arguments(arguments)
-        execute_process(path, args)
+      if powershell_file?(script)
+        mapped_args = arguments.map do |a|
+          "$invokeArgs.ArgumentList += @'\n#{a}\n'@"
+        end.join("\n")
+        # TODO: watch out for File.read encoding - winrm is 65001
+        # TODO: need to cleanup after ScriptBlock? purge shell history?
+        # TODO: document scripts may not include @' '@ style here-strings
```

Comment:
Workarounds for nested here-strings - https://www.reddit.com/r/PowerShell/comments/589fsw/nested_herestring/